### PR TITLE
[Programmes 6900] ATI av events for clips

### PR DESCRIPTION
**Pull request by benolot**
Ticket: https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6900

### CODE REVIEW

File: assets/js/smp/smp.js [hunk 1]
```diff
@@ -100,6 +100,7 @@ define(['jquery-1.9', 'smp/smpStatePersistance', 'bump-3'], function ($, SmpStat
 
         var loadPlayer = function() {
             setStatsFromRecommendations();
+            self.options.smpSettings.statsObject.destination = 'PROGRAMMES_PS_TEST';
```
Comment (benolot) [inline assets/js/smp/smp.js]: This needs setting properly (WIP), but still want feedback on all overall changes

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php [hunk 2]
```diff
@@ -127,14 +155,30 @@ public function getSmpConfig(): array
         if (!empty($this->analyticsCounterName)) {
             $smpConfig['smpSettings']['counterName'] = $this->analyticsCounterName;
         }
+
+        $brand = $this->programmeItem->getTleo();
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php]: TLEO is not necessarily a brand

File: src/Controller/BaseController.php [hunk 3]
```diff
@@ -230,9 +232,17 @@ protected function resolvePromises(array $promises): array
     protected function renderWithChrome(string $view, array $parameters = [])
     {
         $this->preRender();
-
         $cosmosInfo = $this->container->get(CosmosInfo::class);
+
+        // Didn't iStats die? Why is this still here?
+        // Without this bit, the SMP Stats won't work, until we are happy that we should remove iStats completely
+        // then we can get rid of this.
+        $istatsAnalyticsLabels = new IstatsAnalyticsLabels($this->context, $this->istatsProgsPageType, $cosmosInfo->getAppVersion(), $this->istatsExtraLabels);
```
Comment (BjornTwachtmann) [inline src/Controller/BaseController.php]: Hmm, there's a couple of function that were already doing this, the calls to them were removed in #1028 ... which is already live... nevertheless it makes sense to call those

Comment (benolot) [reply src/Controller/BaseController.php]: I didn't want to call the whole setting up again, so I just took this out of the setupIStatsAndSomething method

File: src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php (hunk 2 above)
Comment (benolot) [reply src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php]: Updated

File: src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php [hunk 4]
```diff
@@ -104,6 +120,16 @@ public function getSmpConfig(): array
             $this->streamableVersion
         );
 
+        $series = null;
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php]: $seriesPid, $episodePid ?

File: src/ValueObject/AtiAnalyticsLabels.php [hunk 5]
```diff
@@ -52,9 +59,19 @@ public function orbLabels()
             $masterbrandCustomVar = urlencode($masterbrandCustomVar);
         }
 
+        $producer = null;
+        if ($this->context) {
+            $producer = $this->producerVariableHelper->calculateProducerVariable($this->context);
+        }
+
+        $destination = 'BBC';
```
Comment (BjornTwachtmann) [inline src/ValueObject/AtiAnalyticsLabels.php]: previous default destination looks to be programmes_ps, why the change?

File: src/ValueObject/AtiAnalyticsLabels.php [hunk 6]
```diff
@@ -52,9 +59,19 @@ public function orbLabels()
             $masterbrandCustomVar = urlencode($masterbrandCustomVar);
         }
 
+        $producer = null;
```
Comment (BjornTwachtmann) [inline src/ValueObject/AtiAnalyticsLabels.php]: previous default producer was "BBC"? Or is my reading comprehension way off today?

File: src/Controller/BaseController.php (hunk 3 above)
Comment (BjornTwachtmann) [reply src/Controller/BaseController.php]: *ahem*

File: src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php (hunk 4 above)
Comment (benolot) [reply src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php]: Changed

File: src/ValueObject/AtiAnalyticsLabels.php (hunk 6 above)
Comment (benolot) [reply src/ValueObject/AtiAnalyticsLabels.php]: Fixed, I got them the wrong way around

File: src/ValueObject/AtiAnalyticsLabels.php (hunk 5 above)
Comment (benolot) [reply src/ValueObject/AtiAnalyticsLabels.php]: Wrong way around, my bad, fixed

File: src/Controller/BaseController.php (hunk 3 above)
Comment (benolot) [reply src/Controller/BaseController.php]: Finally updated

File: src/Controller/Helpers/DestinationVariableHelper.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        $destination =  'programmes_ps';
+
+        if (method_exists($context, 'getNetwork') && $context->getNetwork()
+            && (in_array((string) $context->getNetwork()->getNid(), ['bbc_world_service', 'bbc_world_service_tv', 'bbc_learning_english'])
+                || $context->getNetwork()->isWorldServiceInternational())
+        ) {
+            $destination = 'ws_programmes';
+        }
+
+        if (in_array($appEnvironment, ['int', 'stage', 'sandbox', 'test'])) {
```
Comment (Tasiobg) [inline src/Controller/Helpers/DestinationVariableHelper.php]:
```suggestion
        if (\in_array($appEnvironment, ['int', 'stage', 'sandbox', 'test'])) {
```
Not a required changed but something worth knowing, using fully qualified name is faster, some info https://github.com/Roave/FunctionFQNReplacer

File: src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php [hunk 8]
```diff
@@ -104,6 +120,16 @@ public function getSmpConfig(): array
             $this->streamableVersion
         );
 
+        $seriesPid = null;
+        $episodePid = null;
+        foreach ($this->programmeItem->getAncestry() as $parent) {
+            if ($parent->getType() == 'episode') {
```
Comment (Tasiobg) [inline src/Ds2013/Presenters/Utilities/SMP/SmpPresenter.php]: Not a required change but I think is worth mentioning, using strict comparison is a good practice

File: src/Controller/Helpers/DestinationVariableHelper.php [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Controller\Helpers;
+
+class DestinationVariableHelper
+{
+    public function getDestinationFromContext($context, $appEnvironment)
```
Comment (Tasiobg) [inline src/Controller/Helpers/DestinationVariableHelper.php]:
```suggestion
    public function getDestinationFromContext($context, $appEnvironment): string

**Comment by benolot** [comment]
Test this please
